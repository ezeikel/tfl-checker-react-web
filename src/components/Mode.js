import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import renderOptions from "../utils/renderOptions";

const Wrapper = styled.div`
  display: flex;
  align-items: center;
  border: 1px solid #d9d9d9;
  padding: 8px;
  border-radius: 4px;
  font-size: 1.4rem;
  font-weight: bold;
  span {
    margin-left: 8px;
  }
`;

const convertModeToIcon = mode => {
  const icon = {};

  switch (mode) {
    case "walking":
      icon.name = "walking";
      icon.color = "#34495E";
      break;
    case "bus":
    case "replacement-bus":
      icon.name = "bus";
      icon.color = "#C83638";
      break;
    case "coach":
      icon.name = "bus-alt";
      icon.color = "#10bd59";
      break;
    case "tube":
      icon.name = "subway";
      icon.color = "#34495E";
      break;
    case "national-rail":
      icon.name = "train";
      icon.color = "#34495E";
      break;
    case "overground":
      icon.name = "train";
      icon.color = "#E46B24";
      break;
    case "tflrail":
      icon.name = "train";
      icon.color = "#0026A5";
      break;
    case "dlr":
      icon.name = "tram";
      icon.color = "#26AFAC";
      break;
    default:
      console.error("Mode icon not found.");
  }

  return icon;
};

const Mode = ({
  leg: {
    routeOptions,
    mode: { id },
  },
}) => (
  <Wrapper>
    <FontAwesomeIcon
      icon={["fad", convertModeToIcon(id).name]}
      color={convertModeToIcon(id).color}
      size="lg"
    />
    {renderOptions(routeOptions)}
  </Wrapper>
);

Mode.propTypes = {
  leg: PropTypes.shape({
    mode: PropTypes.shape({
      id: PropTypes.string,
    }),
    routeOptions: PropTypes.arrayOf(PropTypes.shape({})),
  }).isRequired,
};

export default Mode;
