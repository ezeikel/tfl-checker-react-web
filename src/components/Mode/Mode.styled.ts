/* eslint-disable import/prefer-default-export */

import styled from "styled-components";

export const Wrapper = styled.div`
  display: flex;
  align-items: center;
  border: 1px solid #d9d9d9;
  padding: 8px;
  border-radius: 4px;
  font-size: 1.4rem;
  font-weight: bold;
  span {
    margin-left: 8px;
  }
`;
