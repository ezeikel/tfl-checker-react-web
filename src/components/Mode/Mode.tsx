import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { Wrapper } from "./Mode.styled";
import convertModeToIcon from "../../utils/convertModeToIcon";

type ModeProps = {
  leg: {
    mode: {
      id: string;
    };
    routeOptions: any[];
  };
};

const Mode = ({
  leg: {
    routeOptions,
    mode: { id },
  },
}: ModeProps) => {
  const { name, color } = convertModeToIcon(id);

  return (
    <Wrapper>
      <FontAwesomeIcon icon={["fad", name]} color={color} size="lg" />
      {routeOptions.map((option, i) => (
        // eslint-disable-next-line react/no-array-index-key
        <span key={i}>
          {option.name}
          {i < routeOptions.length - 1 ? <span>/</span> : ""}
        </span>
      ))}
    </Wrapper>
  );
};

export default Mode;
