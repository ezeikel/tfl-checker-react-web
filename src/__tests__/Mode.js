import React from 'react';
import { render } from '@testing-library/react';
import Mode from '../components/Mode';

let mockProps;

beforeEach(() => {
  mockProps = {
    leg: {
      mode: {
        id: "walking"
      },
      routeOptions: [
        {

        }
      ]
    }
  };
});

test('renders without crashing', () => {
  const { container } = render(
    <Mode {...mockProps} /> // eslint-disable-line react/jsx-props-no-spreading
  );
  expect(container).toMatchSnapshot();
});
