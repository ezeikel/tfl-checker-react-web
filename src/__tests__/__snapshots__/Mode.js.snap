// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div>
  <div
    class="sc-AxjAm hoPHcu"
  >
    <svg
      class="svg-inline--fa fad-walking fa-lg"
      color="#34495E"
    />
    <span>
      
    </span>
  </div>
</div>
`;
